### DISCUSSION TITLE
Make memo optionally required on crypto transfers

### DISCUSSION THREAD

**Discussion by gregscullard** [hiero-ledger/hiero-improvement-proposals#81] [Ideas]
A number of exchanges require a memo on transfers from user accounts to the exchange. Often this step is missed, resulting in the transfer succeeding but not being allocated to the appropriate exchange user account.

As a result, the user who initiated the transfer has to contact the exchange, prove the transfer occurred by way of a transaction id or hash so that the exchange can verify the transaction and finally allocate it to the user's exchange account which may be a time consuming process.

The idea here is to add a `memo required on payments` true/false flag on accounts such that if enabled (true), a crypto transfer to the account will fail with an error (`MEMO_REQUIRED` for example) if no memo was supplied. The likelihood is that only exchange accounts would set this flag to true, however there may be other use cases where this feature could be useful.

This would likely catch 99% of the failures to input a memo, the remaining 1% may be mistyped memos.

Don't hesitate to comment on the usefulness of this feature for other entities (files, contracts, tokens, topics) or transaction types so that an eventual HIP takes those into consideration.

**Comment by kenthejr** [comment]
I like this idea. Do you think requiring memo is explicit enough or would it make more sense to have a routing protocol (like a "route" field or maybe something explicit as part of the address that gets parsed into the memo field)?

**Comment by bugbytesinc** [comment]
Do we have examples of other DLT's having this feature?  The burden will be on wallet makers to provide a useful error message, I'm assuming this would be a PRECHECK exception if thrown?

**Reply by EvanTedesco** [reply]
[Here](https://xrpl.org/require-destination-tags.html#require-destination-tags) is a link to XRP's RequireDest tag.  [EOS also has this functionality](https://github.com/eosauthority/incoming-transfer-rules)

**Comment by gregscullard** [comment]
Hi @bugbytesinc, I'm not aware of any other network doing this, but I know it's a major pain for users who forget it and likely a draw on exchange resources in terms of support.
I'd expect a new PRECHECK error indeed. Wallets may not handle it "nicely" to start with but at least the transaction will not be allowed through the gates.

@kenthejr not entirely sure, transferring to an exchange using a memo can be done from any address, the exchange doesn't really care about the source of the funds, only the memo. I suspect tying the two together would be more hassle for the entire ecosystem.
